PXE server installation playbook

bluecmd opened the pull request:
Introduce playbooks meant to assist an operator to get started with
Tateru by installing the PXE server with all the needed data.

bluecmd committed:
PXE server installation playbook

Introduce playbooks meant to assist an operator to get started with
Tateru by installing the PXE server with all the needed data.

See https://github.com/tateru/tateru-installer/pull/6 for longer
reasoning why the copytoram size is specified.

fyx commented:
I think DHCP option 129 might be better choice, IANA has it as example for Kernel options:
```
128 | PXE - undefined (vendor specific) [RFC4578]
128 | Etherboot signature. 6 bytes: E4:45:74:68:00:00
128 | DOCSIS "full security" server IP address
128 | TFTP Server IP address (for IP Phone software load)
129 | PXE - undefined (vendor specific) [RFC4578]
129 | Kernel options. Variable length string
129 | Call Server IP address
130 | PXE - undefined (vendor specific) [RFC4578]
130 | Ethernet interface. Variable length string.
130 | Discrimination string (to identify vendor)
131 | PXE - undefined (vendor specific) [RFC4578]
131 | Remote statistics server IP address
132 | PXE - undefined (vendor specific) [RFC4578]
132 | IEEE 802.1Q VLAN ID
133 | PXE - undefined (vendor specific) [RFC4578]
133 | IEEE 802.1D/p Layer 2 Priority
134 | PXE - undefined (vendor specific) [RFC4578]
134 | Diffserv Code Point (DSCP) for VoIP signalling and media streams
135 | PXE - undefined (vendor specific) [RFC4578]
135 | HTTP Proxy for phone-specific applications
```
https://www.iana.org/assignments/bootp-dhcp-parameters/bootp-dhcp-parameters.xhtml

bluecmd commented:
Ok, sounds good - will add svc= override as generic kernel param override on 129, and move tateru-ipxe to use option 130 (or 135? the last one iPXE requests) instead for config. For DHCPv6 maybe `DHCPV6_BOOTFILE_PARAM ` ?

But what DHCPv6 option for kernel params?

EDIT: In light of this, we will use 129 for ipxe config as it is now, and DHCPV6_BOOTFILE_PARAM for DHCPv6 - and nothing for kernel args. You need special kernel args? Add it to the ansible ipxe playbook!

bluecmd commented:
Ready for review @fyx @ventris!

bluecmd committed:
Add Tateru boot service readme

bluecmd committed:
Decrease loglevel to errors by default

Docker emits an overlayfs warning on boot, this makes it quiet.
It seems reasonable anyhow, this is not the production kernel the users
will end up running in production so it does not matter.

ventris commented on docs/boot-server/ipxe.cfg.j2:
```diff
@@ -0,0 +1,44 @@
+cpair --foreground 7 --background 9 2
+cpair --foreground 6 --background 9 3
+
+console --picture http://{{ inventory_hostname }}/tateru/ipxe.png
```
We should probably not use inventory_hostname here but instead ansible_fqdn or get the actual IP from gather facts.

ventris commented on docs/boot-server/ipxe.cfg.j2:
```diff
@@ -0,0 +1,44 @@
+  exit
+
+:installer
+  initrd http://{{ inventory_hostname }}/tateru/arch/boot/intel-ucode.img
+  initrd http://{{ inventory_hostname }}/tateru/arch/boot/amd-ucode.img
+  initrd http://{{ inventory_hostname }}/tateru/arch/boot/x86_64/initramfs-linux.img
+  kernel http://{{ inventory_hostname }}/tateru/arch/boot/x86_64/vmlinuz-linux ${kp}
```
We should probably not use inventory_hostname here but instead ansible_fqdn or get the actual IP from gather facts.

ventris commented on docs/boot-server/ipxe.cfg.j2:
```diff
@@ -0,0 +1,44 @@
+set ipparam BOOTIF=${netX/mac} ip=${ip}::${gateway}:${netmask}
+set kp initrd=amd-ucode.img initrd=intel-ucode.img initrd=initramfs-linux.img
+set kp ${kp} loglevel=3 cow_spacesize=768M archisobasedir=arch
+set kp ${kp} archiso_http_srv=http://{{ inventory_hostname }}/tateru/ verify=n
```
We should probably not use inventory_hostname here but instead ansible_fqdn or get the actual IP from gather facts.

ventris reviewed (commented):
inventory_hostname can be an issue due to the fact that the hostname that is set in ansible inventory does not have to be the one the actual server has. You can specify anything in the hostname and then you can specify an ansible_ip fact about that host with the correct ip.
Some people might get caught in this if they use the hostname field in a different than IP or FQDN.
There is a fact called ansible_fqdn that is taken from the host when the Gather Facts step is run. Either that or we just go for the IP of the host based from the same Gather Facts step.

bluecmd committed:
Introduce TATERU_BOOT_URL

This variable controlls how the URL is constructed for loading the
installer. It allows setting to http/https and either IP addresses or
even DNS if those are configured.

bluecmd commented:
@ventris Done, PTAL
